spirv_reflect.h

```diff
@@ -393,7 +393,7 @@ typedef struct SpvReflectTypeDescription {
   const char*                       type_name;
   // Non-NULL if type is member of a struct
   const char*                       struct_member_name;
-  SpvStorageClass                   storage_class;
+  int                               storage_class;
```

Comment:
I would rather try and keep the interface the same... maybe the answer is to use `StorageClassMax` and typedef it as 

`SpvStorageClass invalid_storage_class = StorageClassMax`

where we are purposely setting it to be invalid

Reply:
I guess nvm, this would be wrong because it would show everything

Reply:
maybe what we need is 

```suggestion
// If an invalid Storage Class will be set to -1 because SpvStorageClass has no "invalid" value
  int                               storage_class;
```

Reply:
Yup, I don't think using `Max` is a good idea because its value will potentially change everytime the spirv headers are updated. I don't know if that create a problem for tooling using spirv-reflect.

The other possibility is to change the header file to spirv.hpp11, which uses enum classes. I believe that would make it okay to use `-1`. However, I don't know how much we want to change.

Reply:
thinking more, I think the best we can do is have this as an `int` but leave a comment above each line (as we might introduce compiler warning/errors with this PR) that `storage_class` and `built_in` can be casted to `SpvStorageClass/SpvBuiltIn` if the `int` value is not `-1`

Reply:
The enum class would not work, some of the files are written in C.